src/celeritas/ext/detail/HitProcessor.cc

```diff
@@ -219,18 +189,8 @@ HitProcessor::HitProcessor(SPConstVecLV detector_volumes,
         p->SetPolarization(G4ThreeVector());
     }
 
-    // Create track if user requested particle types
-    for (G4ParticleDefinition const* pd : particles)
-    {
-        CELER_ASSERT(pd);
-        auto track = std::make_unique<G4Track>(
-            new G4DynamicParticle(pd, G4ThreeVector()), 0.0, G4ThreeVector());
-        track->SetTrackID(0);
-        track->SetParentID(0);
-        track->SetStep(step_.get());
-
-        tracks_.emplace_back(std::move(track));
-    }
+    // Set the step for all tracks managed by TrackProcessor
+    track_processor_.set_step_for_tracks(step_.get());
```

Comment:
The step should be part of the TrackProcessor constructor, not set after the fact (splits initialization in to multiple steps, easy to forget). Either defer construction of the track processor (using unique_ptr/shared_ptr/optional) or have the TrackProcessor own the corresponding step (which I think makes more sense).

Reply:
Moved the step to the `TrackProcessor`